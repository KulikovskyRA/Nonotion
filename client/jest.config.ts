const config = {
  verbose: true,
  preset: 'ts-jest',
  testEnvironment: 'jest-environment-jsdom',
  transform: {
    '^.+\\.(ts|tsx|js|jsx)?$': 'ts-jest',
  },
  moduleNameMapper: {
    '\\.(gif|ttf|eot|svg|png)$': '<rootDir>/src/test/__ mocks __/fileMock.js',
  },
};

export default config;
