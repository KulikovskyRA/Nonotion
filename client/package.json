{
  "name": "client",
  "private": true,
  "version": "0.0.0",
  "type": "module",
  "scripts": {
    "dev": "vite",
    "build": "tsc && vite build",
    "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
    "preview": "vite preview",
    "test": "jest"
  },
  "dependencies": {
    "@reduxjs/toolkit": "^1.9.7",
    "@types/react-redux": "^7.1.28",
    "antd": "^5.10.3",
    "dotenv": "^16.3.1",
    "jest-environment-jsdom": "^29.7.0",
    "lodash": "^4.17.21",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-redux": "^8.1.3",
    "react-router-dom": "^6.18.0",
    "vite-plugin-environment": "^1.1.3"
  },
  "devDependencies": {
    "@babel/plugin-transform-modules-commonjs": "^7.23.3",
    "@babel/preset-env": "^7.23.3",
    "@testing-library/dom": "^9.3.3",
    "@testing-library/jest-dom": "^6.1.4",
    "@testing-library/react": "^14.1.0",
    "@testing-library/user-event": "^14.5.1",
    "@types/jest": "^29.5.8",
    "@types/lodash": "^4.14.200",
    "@types/react": "^18.2.15",
    "@types/react-dom": "^18.2.7",
    "@types/react-test-renderer": "^18.0.6",
    "@typescript-eslint/eslint-plugin": "^6.0.0",
    "@typescript-eslint/parser": "^6.0.0",
    "@vitejs/plugin-react": "^4.0.3",
    "babel-jest": "^29.7.0",
    "eslint": "^8.45.0",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-react-refresh": "^0.4.3",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.7.0",
    "react-test-renderer": "^18.2.0",
    "sass": "^1.69.5",
    "ts-jest": "^29.1.1",
    "ts-node": "^10.9.1",
    "typescript": "^5.0.2",
    "vite": "^4.4.5"
  }
}
