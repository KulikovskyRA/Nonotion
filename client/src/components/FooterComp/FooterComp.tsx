import { Footer } from 'antd/es/layout/layout';

const FooterComp = () => {
  console.log('Footer - rendering');
  return (
    <Footer
      style={{
        backgroundColor: '#001529',
        textAlign: 'center',
        color: 'white',
        padding: 10,
      }}
    >
      Nonotion Â©2023 Created by Kulikovsky
    </Footer>
  );
};

export default FooterComp;
