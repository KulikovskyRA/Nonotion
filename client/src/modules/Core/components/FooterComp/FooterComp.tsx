import { Footer } from 'antd/lib/layout/layout';

const FooterComp = () => {
  console.log('Footer - rendering');
  return (
    <Footer
      data-testid="Footer"
      style={{
        backgroundColor: 'white',
        textAlign: 'center',
        color: 'black',
        padding: 10,
      }}
    >
      Nonotion Â©2023 Created by Kulikovsky
    </Footer>
  );
};

export default FooterComp;
