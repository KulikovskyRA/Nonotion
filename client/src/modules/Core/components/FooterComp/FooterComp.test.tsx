import { createRoot } from 'react-dom/client';

import FooterComp from './FooterComp';

import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

describe('Проверка компонента FooterComp', () => {
  afterEach(cleanup);

  test('Не крашится', () => {
    const div = document.createElement('div');
    const root = createRoot(div);
    root.render(<FooterComp />);
  });

  test('Отрисовывает текст в футере', () => {
    const { getByTestId } = render(<FooterComp></FooterComp>);
    expect(getByTestId('Footer')).toHaveTextContent(
      'Nonotion ©2023 Created by Kulikovsky'
    );

    const footerText = screen.getByText(/Nonotion/i);
    expect(footerText).toBeInTheDocument();
  });

  test('Снапшот тестирование', () => {
    const tree = renderer.create(<FooterComp></FooterComp>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
