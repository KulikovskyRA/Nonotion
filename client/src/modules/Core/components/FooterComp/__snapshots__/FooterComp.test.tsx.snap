// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка компонента FooterComp Снапшот тестирование 1`] = `
<footer
  className="ant-layout-footer css-dev-only-do-not-override-qgg3xn"
  data-testid="Footer"
  style={
    {
      "backgroundColor": "#001529",
      "color": "white",
      "padding": 10,
      "textAlign": "center",
    }
  }
>
  Nonotion ©2023 Created by Kulikovsky
</footer>
`;
